File: UI-OC/UISteper-oc/UISteper-oc/ViewController.m
//
//  ViewController.m
//  UISteper-oc
//
//  Created by Rocky on 2018/7/4.
//  Copyright © 2018年 Rocky. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UILabel *mylabel;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    UIStepper* stepper = [[UIStepper alloc] initWithFrame:CGRectMake(140, 122, 94, 29)];
    // 设置最小值
    stepper.minimumValue = 1;
    // 设置最大值
    stepper.maximumValue = 99;
    // 设置步长
    stepper.stepValue = 3;
    // 设置颜色
    [stepper setTintColor:[UIColor redColor]];
    // 设置背景色
    [stepper setBackgroundColor:[UIColor cyanColor]];
    //首尾相连 min <-> max
    stepper.wraps = YES;
    //按住时自增或自减
    stepper.autorepeat = YES;
    //连续的
    stepper.continuous = NO;
    // 关联target-action
    [stepper addTarget:self action:@selector(stepperAction:) forControlEvents:UIControlEventValueChanged];
}

- (IBAction)stepperAction:(id)sender {
    UIStepper* step = (UIStepper*)sender;
    // 获取value
    CGFloat value = step.value;
    // 显示到label上
    NSString* text = [NSString stringWithFormat:@"%g",value];
    self.mylabel.text = text;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: UI-Swift/IOSUI-2/IOSUI-2/ViewController.swift
//
//  ViewController.swift
//  IOSUI-2
//
//  Created by Rocky on 2018/6/21.
//  Copyright © 2018年 Rocky. All rights reserved.
//

import UIKit

class ViewController: UIViewController , UITextFieldDelegate{

    override func viewDidLoad() {
        super.viewDidLoad()
        
        let label = UILabel(frame: CGRect(x: 40, y: 100, width: 100, height: 30))
        label.text = "0"
        label.tag = 1
        self.view.addSubview(label)
        
        let stepper = UIStepper(frame: CGRect(x: 90, y: 100, width: 0, height: 0))
        stepper.stepValue = 1
        stepper.sizeToFit()
        stepper.value = 0
        stepper.minimumValue = 1
        stepper.maximumValue = 10000
        stepper.addTarget(self, action: #selector(ViewController.stepperAction( _ :)), for: UIControlEvents.valueChanged)
        self.view.addSubview(stepper)
        
        let input = UITextField(frame: CGRect(x: 90, y: 150, width: 200, height: 50))
        input.borderStyle = UITextBorderStyle.roundedRect
        input.placeholder = "your name"
        input.autocorrectionType = UITextAutocorrectionType.no
        input.returnKeyType = UIReturnKeyType.done
        input.clearButtonMode = UITextFieldViewMode.whileEditing
        input.keyboardType =  UIKeyboardType.namePhonePad
        input.keyboardAppearance = UIKeyboardAppearance.dark
        input.delegate = self
        self.view.addSubview(input)
        
        let btnAlert = UIButton(frame: CGRect(x: 90, y: 210, width: 150, height: 40))
        btnAlert.setTitle("点击-提示框", for: UIControlState())
        btnAlert.backgroundColor = UIColor.lightGray
        btnAlert.addTarget(self, action: #selector(ViewController.alertAction(_ :)), for: UIControlEvents.touchUpInside)
        self.view.addSubview(btnAlert)
        
        let dataPicker = UIDatePicker()
        dataPicker.center = CGPoint(x: 180, y: 400)
        dataPicker.tag = 11//biaoshi
        self.view.addSubview(dataPicker)
        let btnData = UIButton(frame: CGRect(x: 90, y: 550, width: 150, height: 40))
        btnData.setTitle("Data", for: UIControlState())
        btnData.backgroundColor = UIColor.lightGray
        btnData.addTarget(self, action: #selector(ViewController.getData(_ :)), for: UIControlEvents.touchUpInside)
        self.view.addSubview(btnData)
    }
    @objc func getData(_ ui: UIButton){
        let dataPicker = self.view.viewWithTag(11) as? UIDatePicker
        let date = dataPicker?.date
        let dataFormaat = DateFormatter()
        dataFormaat.dateFormat = "yyyy-MM-dd HH:mm"
        let dataTime = dataFormaat.string(from: date!)
        let alert = UIAlertController(title: "当前日期", message: dataTime, preferredStyle: UIAlertControllerStyle.alert)
        let action = UIAlertAction(title: "OK", style: UIAlertActionStyle.cancel, handler: nil)
        
        alert.addAction(action)
        
        self.present(alert,animated: true, completion: nil)
    }
    @objc func alertAction( _ ui: UIButton){
        let alert = UIAlertController(title: "choose", message: "选择打开方式", preferredStyle: UIAlertControllerStyle.actionSheet)
        let action = UIAlertAction(title: "相机", style: UIAlertActionStyle.default, handler: {
            (alert :UIAlertAction) in print("点击了相机")
        })
        let action1 = UIAlertAction(title: "相册", style: UIAlertActionStyle.default, handler: {
            (alert: UIAlertAction) in print("点击了相册")
        })
        let cancel = UIAlertAction(title: "取消", style: UIAlertActionStyle.cancel, handler: nil)
        
        alert.addAction(action)
        alert.addAction(action1)
        alert.addAction(cancel)
        
        self.present(alert, animated: true,completion: nil)
    }
    
    func textFieldShouldReturn( _ text: UITextField) -> Bool {
        text.resignFirstResponder()
        return true
    }
    
    @objc func stepperAction(_ step : UIStepper){
        let value = step.value
        let label = self.view.viewWithTag(1) as! UILabel
        label.text = "\(Int(value))"
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

